src/sdk/index.js

```diff
@@ -4,23 +4,42 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = {...globalSpec.headers()}
+  }
+  else {
+    headers = globalSpec.headers
+  }
+
+  return {
+    ...globalSpec,
+    ...{headers}
+  }
+}
+
 export default function sdk (globalSpec = {}) {
 
   let defaultSpec = {
     origin: '',
     headers: DEFAULT_HEADERS
   }
-  let options = merge({}, defaultSpec, globalSpec)
+
+  let options = merge({}, defaultSpec, configGlobalSpec(globalSpec))
+
   let stream = Observable.create()
 
   function apiFactory (instanceSpec = {}) {
 
-  	return api(merge({}, options, instanceSpec, {
```

Comment:
I'm confused because `x-entity` and `authorization` are the headers that the getHeaders function returns. However, we don't want to lose other things passed to globalSpec...which we aren't because any property on globalSpec that isn't headers doesn't get overwritten because of `let options = merge({}, defaultSpec, globalSpec)`